src/main/java/seedu/address/model/task/Note.java

```diff
@@ -0,0 +1,68 @@
+package seedu.address.model.task;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Objects;
+
+/**
+ * Represents a Task's note in the task list.
+ * Guarantees: immutable; is valid as declared in {@link #isValidNote(String)}
+ */
+public class Note {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Note should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * The first character of the task note must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final String taskNote;
```

Comment:
```suggestion
    public final String value;
```

Perhaps a simpler name will do. Not a fan of parent referencing.

For reference, AB3 does this too

![image](https://github.com/AY2324S1-CS2103T-T10-2/tp/assets/65714641/96f7ef65-5276-498c-ae43-2b47e18b44ef)

https://github.com/AY2324S1-CS2103T-T10-2/tp/blob/master/src/main/java/seedu/address/model/person/Address.java

Reply:
Resolved in https://github.com/AY2324S1-CS2103T-T10-2/tp/pull/81/commits/e92f537770862854c44c9f42c63beebf730d0639